CMakeLists.txt

```diff
@@ -1,42 +1,123 @@
+  ${NO_WARN_UNUSED_LAMBDA_CAPTURE} ${NO_WARN_MAYBE_UNINITIALIZED} ${NO_WARN_MISMATCHED_TAGS} ${NO_WARN_INFINITE_RECURSION})  
+
 file(GLOB RACK_SOURCES src/*.cpp)
-set(RACK_DIR "/Users/paul/dev/music/rack/Rack-SDK")
+list(REMOVE_ITEM RACK_SOURCES src/SurgeXT.cpp)
+
+add_compile_options("-fvisibility=hidden" "-fvisibility-inlines-hidden")
 
 add_library(${PROJECT_NAME} STATIC ${RACK_SOURCES})
-target_include_directories(${PROJECT_NAME} PRIVATE
-        ${RACK_DIR}/include
-        ${RACK_DIR}/dep/include
-        )
 target_link_libraries(${PROJECT_NAME} PRIVATE surge-common)
 
-add_custom_command(TARGET ${PROJECT_NAME}
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/dep/surge-build
-        COMMAND RACK_DIR=${RACK_DIR} arch -x86_64 make -j surge-common
-        USES_TERMINAL)
+add_library(${LIB_NAME} MODULE src/SurgeXT.cpp)
+target_compile_options(${LIB_NAME} PRIVATE "-fuse-ld=gold" "-flto")
+
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+  set_target_properties(${LIB_NAME} PROPERTIES SUFFIX ".dylib")
+  set_target_properties(${LIB_NAME} PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
+endif ()
+
+set_target_properties(${LIB_NAME} PROPERTIES PREFIX "")
 
-add_custom_command(TARGET ${PROJECT_NAME}
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-        COMMAND RACK_DIR=${RACK_DIR} arch -x86_64 make -j
-        COMMAND ${CMAKE_COMMAND} -E copy plugin.json /Users/paul/Documents/Rack2/plugins/SurgeXTRack
-        COMMAND ${CMAKE_COMMAND} -E copy plugin.dylib /Users/paul/Documents/Rack2/plugins/SurgeXTRack
-        USES_TERMINAL)
+if (${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+  target_link_libraries(${LIB_NAME} PRIVATE ${PROJECT_NAME} surge-common juce_dsp_rack_sub tinyxml strnatcmp
+    sst-plugininfra fmt sqlite oddsound-mts eurorack airwindows winmm uuid wsock32 shlwapi version wininet ole32 ws2_32)
+endif ()
 
+install(TARGETS ${LIB_NAME} surge-common juce_dsp_rack_sub tinyxml strnatcmp sst-plugininfra fmt sqlite oddsound-mts eurorack airwindows
+        RUNTIME DESTINATION bin
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib/static)
 
-# This is the full build
-add_custom_target(${PROJECT_NAME}_FULL)
-add_dependencies(${PROJECT_NAME}_FULL ${PROJECT_NAME})
-add_custom_command(TARGET ${PROJECT_NAME}_FULL
-        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-        COMMAND RACK_DIR=${RACK_DIR} arch -x86_64 make dist -j
-        COMMAND RACK_DIR=${RACK_DIR} arch -x86_64 make install -j
-        USES_TERMINAL)
+file(GLOB LICENSE LICENSE*)
+file(INSTALL plugin.json res docs patches presets ${LICENSE} README.md DESTINATION ${PLUGIN_NAME})
+file(INSTALL surge/resources/surge-shared/configuration.xml
+             surge/resources/surge-shared/windows.wt
+             surge/resources/data/wavetables
+             surge/resources/data/wavetables_3rdparty
+             surge/resources/data/fx_presets
+             res/surge_extra_data/fx_presets
+    DESTINATION ${PLUGIN_NAME}/build/surge-data)
```

Comment:
Is the folder `build/surge-data` correct or shouldn't this just be `surge-data`? Is this folder used/referenced by source code?